src/main/java/net/datafaker/providers/base/Medical.java

```diff
@@ -30,11 +45,19 @@ public String diagnosisCode() {
         return faker.regexify(regex);
     }
 
+    /**
+     * @deprecated since 2.4.0. For removal in 3.0.0 version. Use {@link net.datafaker.providers.healthcare.MedicalProcedure#icd10()} instead.
+     */
+    @Deprecated(since = "2.4.0", forRemoval = true)
     public String procedureCode() {
         String regex = resolve("medical.procedure_code.icd10");
         return faker.regexify(regex);
     }
 
+    /**
+     * @deprecated since 2.4.0. For removal in 3.0.0 version. Use {@link net.datafaker.providers.healthcare.CareProvider#medicalProfession()} instead.
```

Comment:
I probably wouldn't add the "for removal in". The 3.0.0 can be released next week, or it can be done 5 years from now, that's hard to say, we usually keep these things around 1-2 released, and then remove them.

Reply:
Ok, adjusted.